menas/ui/components/schema/schemaDetail.controller.js

```diff
@@ -221,6 +295,12 @@ sap.ui.define([
         this._schemaRestDAO.getLatestVersionByName(currentSchema.name)
           .then(version => this._model.setProperty("/editingEnabled", currentSchema.version === version));
       }
+    },
+
+    isHttpUrl: function (url) {
+      if (!url) return false;
+      const pattern = new RegExp('^https?://[^ ]+$', 'i');
```

Comment:
I see what you did. But what about making it more general? 
```js
var pattern = new RegExp('^(https?:\\/\\/)?'+ // protocol
    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // domain name
    '(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)|'+ // OR IPv4 address
    '((?:[a-f0-9]{1,4}:){7}[a-f0-9]{1,4}))' + // OR IPv6 address
    '(\\:\\d+)?' + // port
    '(\\/[-a-z\\d%_.~+]*)*'+ // path
    '(\\?[;&a-z\\d%_.~+=-]*)?'+ // query string
    '(\\#[-a-z\\d_]*)?$','i'); // fragment locator
```

Reply:
I did find this, too. I even considered using some well-known URL regexes from https://mathiasbynens.be/demo/url-regex. However, they all seem to have one important downside for us, they only seem to allow public domains; things like http://localhost/something could cause problems. Which is why did not want to include something this advanced and they butcher it with dev extensions 😆 

So, sure, I could, just let's be aware of the limitations...

Reply:
`localhost` is `127.0.0.1` which is allowed by my solution as well as `::1`
You can implement a different one as well. The website you provided is quite nice!

Reply:
I like the pattern provided by Saša. But I'd suggest to do it in a separate task/PR and implement it in a separate reusable component, covered by unit tests etc, to make sure everything we want to pass validation actually passes.

In this PR the simple RegExp is good for now. It really helps if a user mistypes an URL and clicks 'Submit URL'. And if the URL is wrong the user will get the message from the backend in a popup message box.

Reply:
OK, related issue created: https://github.com/AbsaOSS/enceladus/issues/1250